### PR TITLE
Change npc indicator plugin to not edit npc composition

### LINKED ISSUE #3844: Cannot shift tag menaphite thugs
Cannot right click shift tag menaphite thugs (useful for blackjacking) but it is possible to add the NPC to the NPC tagging list by hand; however, this is undesirable since it tags all 50 thugs in the nearby location.

### LINKED ISSUE #4206: NPC Indicator's Tag Is Not Appearing on Some NPCs
NPC Indicator's Tag option doesn't appear on some NPCs like [Torfinn](http://oldschoolrunescape.wikia.com/wiki/Torfinn), Bandits and Villagers in [Pollnivneach](http://oldschoolrunescape.wikia.com/wiki/Pollnivneach). I think it's because there's too many menu options, because that the only correlation I see.

<details>
 <summary>stack trace</summary>
 
 ```
  2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Menaphite Thug (3550)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6138) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=0
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Villager (3557)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6140) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=0
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Ali the Camel (3543)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Ali the Camel (3543) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=0
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Ali the Camel (3543)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Ali the Camel (3543) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Menaphite Thug (3550)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6138) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Monkey (23)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Monkey (23) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=1
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Menaphite Thug (3550)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6138) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Rug Merchant (19)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Rug Merchant (19) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=1
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Menaphite Thug (3550)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6138) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Villager (3554)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6139) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=0
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Menaphite Thug (3550)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6138) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Menaphite Thug (3550)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6138) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Menaphite Thug (3550)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6138) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Snake (3545)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Snake (3545) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=0
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Ali the Snake Charmer (3542)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Ali the Snake Charmer (3542) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=1
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Street urchin (3539)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Street urchin (3539) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=1
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Villager (3560)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6141) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=0
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Villager (3557)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6140) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Drunken Ali (3534)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Drunken Ali (3534) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=1
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Ali The barman (3535)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Ali The barman (3535) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=1
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Aristarchus (3569)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Aristarchus (3569) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=1
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Menaphite Thug (3550)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6138) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Villager (3554)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6139) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Menaphite Thug (3550)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: null (6138) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i].equals(npcOption) - i=0, Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - Crocodile (4184)
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - AddNPCOption: Crocodile (4184) - Tag
2018-07-06 22:31:48 [AWT-EventQueue-0] DEBUG n.runelite.client.menus.MenuManager - actions[i] == null && unused == -1 - i=0

 ```
</details>

[Shows for the Rug Merchant](https://i.imgur.com/RoJzK6a.png)
[Doesn't show for Menaphite Thug](https://i.imgur.com/QFysJLw.png)

[First portion is the method I added the debug statements to try and figure out what was going on. The second portion is the output. This hastebin is also provided within the stacktrace above.](https://hastebin.com/exocoyasos.java)

### LINKED ISSUE #4676: Left-click option for Kraken Boss Whirlpools become "Tag" after tagging the Kraken Boss
This is without holding down shift, and after disabling the NPC tagging option.

### PR DESCRIPTION
**Pull request by MagicfTail**
Fixes the tag option staying on npc's if they despawn while holding shift

Closes #4206
Closes #3844
Closes #4676

Note, this moves the tag option below walk here
https://i.imgur.com/Q9MT3D1.png

### CODE REVIEW

**Comment by MagicfTail** [comment]
Closes #4206
Closes #3844
Closes #4676

**Comment by Adam-** [comment]
What happened to `config.isTagEnabled()` here?

File: runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java [hunk 1]
```diff
@@ -241,45 +238,63 @@ public void onConfigChanged(ConfigChanged configChanged)
...
+		menuEntry.setParam1(event.getActionParam1());
+		menuEntry.setIdentifier(event.getIdentifier());
+		menuEntry.setType(MenuAction.RUNELITE.getId());
+		client.setMenuEntries(menuEntries);
+	}
+
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked click)
+	{
+		if (!click.getMenuOption().equals(TAG) || click.getMenuAction() != MenuAction.RUNELITE)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java]: The menu action check here is faster, do it first.

**Comment by MagicfTail** [comment]
The `config.isTagEnabled()` check has been added back.

**Comment by Nightfirecat** [comment]
Not a big fan of the placement of the tag option personally. I'm sure I could learn to live with it, but it would be nice if it could remain the second menu option.

File: runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java [hunk 2]
```diff
@@ -241,45 +238,63 @@ public void onConfigChanged(ConfigChanged configChanged)
 	@Subscribe
 	public void onFocusChanged(FocusChanged focusChanged)
 	{
-		// If you somehow manage to right click while holding shift, then click off screen
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java]: This comment probably shouldn't be removed.

**Comment by MagicfTail** [comment]
It probably could. But I think placing our custom menu options out of the way so they don't change the expected layout too much is better.

**Comment by Nightfirecat** [comment]
That's understandable, but I feel changing what has now been long-standing behavior should have a better reason than "it's technically neater". Just my two cents though

File: runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java (hunk 2 above)
Comment (MagicfTail) [reply runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java]: It doesn't apply any more due to the right click menu closing when the client loses focus. It's also now very obvious what's happening.

File: runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java [hunk 3]
```diff
@@ -238,22 +240,41 @@ public void onConfigChanged(ConfigChanged configChanged)
...
 	public void onMenuOptionClicked(MenuOptionClicked click)
 	{
-		if (!config.isTagEnabled())
+		if (click.getMenuAction() != MenuAction.RUNELITE || !click.getMenuOption().equals(TAG))
 		{
 			return;
 		}
 
-		if (click.getMenuOption().equals(TAG) && NPC_MENU_ACTIONS.contains(click.getMenuAction()))
+		if (click.getMenuOption().equals(TAG) && click.getMenuAction() == MenuAction.RUNELITE)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/npchighlight/NpcIndicatorsPlugin.java]: This entire expression is always true

**Pull request merged**